### PR TITLE
Added Ibex as a target (under Verilator)

### PR DESCRIPTION
**Pull request by eroom1966**
Also modified the prolog code for targets not supporting supervisor mode
certain CSR's are only written if privilege modes are supported

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,4 +1,8 @@
-2016-06-21 Ben Selfridge <benselfridge@galois.com>
+2019-08-08 Lee Moore <moore@imperas.com>
+    * Added support for lowRISC/ibex RTL as a target using Verilator.
+      In conjunction with Philipp Wagner of lowRISC imphil@philipp-wagner.com
```
Comment (asb) [inline ChangeLog]: Probably better to use Philipp's phw@lowrisc.org work email

Comment (eroom1966) [reply ChangeLog]: thanks updated in new pull request

**Comment by eroom1966** [comment]
> I suspect this needs breaking into two pull requests, one for Ibex and
> one for some general tidying up. Could you let me know your thoughts
> on the above and I'll then review again.

No problem, let me take another look

Thx
Lee

**Comment by eroom1966** [comment]
need to resubmit

**Pull request closed without merging**